Common/Orders/Fills/EquityFillModel.cs

```diff
@@ -0,0 +1,634 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Orders.Fees;
+using QuantConnect.Securities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Orders.Fills
+{
+    /// <summary>
+    /// Implements <see cref="IFillModel"/> for Equity
+    /// </summary>
+    public class EquityFillModel : FillModel
```

Comment:
This is very similar to a Fill Model that we built for another product type, so maybe the term "EquityFillModel" should be changed to describe how the model fills the orders?

Reply:
Hey @adam-may! Curious which product type? The current plan as seen in the `DefaultBrokerageModel` is to create a fill model per security type but they could potentially inherit from each other is it's correct